app/src/main/java/com/hillnerds/soundsparrow/Sound.java

```diff
@@ -136,114 +154,34 @@ public void safeSleep(int time) {
         }
     }
 
-    public ArrayList<Note> generateMidiChannel(Channel c) {
-        ArrayList<Note> noteArray = new ArrayList<>();
-
-        int timestampCounter = 0;
-
-        c.generateInstrument(randomGenerator);
-
-        Note instrument_change = new Note(192 + c.number, c.instrument, timestampCounter);
-        noteArray.add(instrument_change);
-
-        for (int i = 0; i < 16; i = i+2) {
-            int startingCode = SoundGeneration.generate_starting_code(c.number);
-            int pitch = SoundGeneration.generate_pitch(randomGenerator, c.range_min, c.range_max);
-            int velocity = SoundGeneration.generate_velocity(randomGenerator);
-
-            int duration = SoundGeneration.generate_note_duration(randomGenerator);
-
-            Note note_on = new Note(startingCode, pitch, velocity, timestampCounter);
-            timestampCounter = timestampCounter + duration;
-
-            Note note_off = new Note(startingCode, pitch, 0, timestampCounter);
-
-            noteArray.add(note_on);
-            noteArray.add(note_off);
-        }
-        return noteArray;
-    }
 
-    public ArrayList <Note> generate_midi_file(ArrayList<Channel> channelList){
-        ArrayList<ArrayList<Note>> midiToMerge = new ArrayList<>();
 
-        for (Channel c : channelList) {
-            midiToMerge.add(generateMidiChannel(c));
-        }
-
-        ArrayList<Note> combined = new ArrayList<>();
+    public class MidiParser implements Runnable {
+        private ArrayList<MidiSequence> midiSequenceFile;
 
-        for (ArrayList<Note> a_m : midiToMerge){
-            combined.addAll(a_m);
+        public MidiParser(ArrayList<MidiSequence> midiSequenceFile) {
+            this.midiSequenceFile = midiSequenceFile;
         }
 
-        return combined;
-    }
-
-    public ArrayList<Note> sort_midi_array (ArrayList<Note> noteArray){
-        Collections.sort(noteArray,new Comparator<Note>() {
-            @Override
-            public int compare(Note m1, Note m2) {
-                return Integer.valueOf(m1.timestamp).compareTo(m2.timestamp);
-            }
-        });
-
-        return noteArray;
-    }
+        public void run() {
+            int currentTimestamp = 0;
 
```

Comment:
Your spacing is inconsistent (number of newlines, number of spaces between things etc)